src/jacobian.jl

```diff
@@ -166,11 +176,20 @@ end
 
 @inline function vector_mode_jacobian!(result, f::F, x::SArray{S,V,D,N}) where {F,S,V,D,N}
     ydual = vector_mode_dual_eval(f, x)
-    extract_jacobian!(result, ydual, N)
-    extract_value!(result, ydual)
+    result = extract_jacobian!(result, ydual, N)
+    result = extract_value!(result, ydual)
     return result
 end
 
+@inline function vector_mode_jacobian!(out::ImmutableDiffResult, f::F, x::SArray{S,V,D,N}) where {F,S,V,D,N}
+    ydual = vector_mode_dual_eval(f, x)
+    jout = extract_jacobian(ydual, x)
+    vout = extract_value(ydual, DiffBase.value(out))
+    out = DiffBase.jacobian!(out, jout)
+    out = DiffBase.value!(out, vout)
+    return out
+end
```

Comment:
Isn't this covered by the method defined directly above (`vector_mode_jacobian!(result, f::F, x::SArray{S,V,D,N})`)? What's the reasoning for a new method here?